src/helpers/format.ts

```diff
@@ -1,10 +1,58 @@
 import type { ReactTestRendererJSON } from 'react-test-renderer';
-import prettyFormat, { plugins } from 'pretty-format';
+import prettyFormat, { NewPlugin, plugins } from 'pretty-format';
 
-const format = (input: ReactTestRendererJSON | ReactTestRendererJSON[]) =>
+type FilterNodeByTypeFunction = (
+  propName: string,
+  propValue: unknown,
+  node: ReactTestRendererJSON
+) => boolean;
+
+export type FormatOptions = {
+  filterProps?: FilterNodeByTypeFunction;
+};
+
+const format = (
+  input: ReactTestRendererJSON | ReactTestRendererJSON[],
+  options: FormatOptions = {}
+) =>
   prettyFormat(input, {
-    plugins: [plugins.ReactTestComponent, plugins.ReactElement],
+    plugins: [getCustomPlugin(options.filterProps)],
     highlight: true,
   });
 
+const getCustomPlugin = (filterProps?: FilterNodeByTypeFunction): NewPlugin => {
+  return {
+    test: (val) =>
+      plugins.ReactTestComponent.test(val) || plugins.ReactElement.test(val),
+    serialize: (val, config, indentation, depth, refs, printer) => {
```

Comment:
As mentioned above you should be able to make this impl. easier by just focusing on `ReactTestComponent` part and using `plugins.ReactElement` as-is.